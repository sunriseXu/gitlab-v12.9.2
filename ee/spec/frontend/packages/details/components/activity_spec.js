import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import PackageActivity from 'ee/packages/details/components/activity.vue';
import {
  npmPackage,
  mavenPackage as packageWithoutBuildInfo,
  mockPipelineInfo,
} from '../../mock_data';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('PackageActivity', () => {
  let wrapper;
  let store;

  function createComponent(packageEntity = packageWithoutBuildInfo, pipelineInfo = null) {
    store = new Vuex.Store({
      state: {
        packageEntity,
      },
      getters: {
        packagePipeline: () => pipelineInfo,
      },
    });

    wrapper = shallowMount(PackageActivity, {
      localVue,
      store,
    });
  }

  const commitMessageToggle = () => wrapper.find({ ref: 'commit-message-toggle' });
  const commitMessage = () => wrapper.find({ ref: 'commit-message' });
  const commitInfo = () => wrapper.find({ ref: 'commit-info' });
  const pipelineInfo = () => wrapper.find({ ref: 'pipeline-info' });

  afterEach(() => {
    if (wrapper) wrapper.destroy();
    wrapper = null;
  });

  describe('render', () => {
    it('to match the default snapshot when no pipeline', () => {
      createComponent();

      expect(wrapper.element).toMatchSnapshot();
    });

    it('to match the default snapshot when there is a pipeline', () => {
      createComponent(npmPackage, mockPipelineInfo);

      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('commit message toggle', () => {
    it("does not display the commit message button when there isn't one", () => {
      createComponent(npmPackage, mockPipelineInfo);

      expect(commitMessageToggle().exists()).toBe(false);
      expect(commitMessage().exists()).toBe(false);
    });

    it('displays the commit message on toggle', () => {
      const commitMessageStr = 'a message';
      createComponent(npmPackage, {
        ...mockPipelineInfo,
        git_commit_message: commitMessageStr,
      });

      commitMessageToggle().trigger('click');

      return wrapper.vm.$nextTick(() => expect(commitMessage().text()).toBe(commitMessageStr));
    });
  });

  describe('pipeline information', () => {
    it('does not display pipeline information when no build info is available', () => {
      createComponent();

      expect(pipelineInfo().exists()).toBe(false);
    });

    it('displays the pipeline information if found', () => {
      createComponent(npmPackage, mockPipelineInfo);

      expect(commitInfo().exists()).toBe(true);
      expect(pipelineInfo().exists()).toBe(true);
    });
  });
});
