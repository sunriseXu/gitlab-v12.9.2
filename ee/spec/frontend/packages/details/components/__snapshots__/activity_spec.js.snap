// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackageActivity render to match the default snapshot when no pipeline 1`] = `
<div
  class="mb-3"
>
  <h3
    class="gl-font-size-16"
  >
    Activity
  </h3>
   
  <div
    class="info-well"
  >
    <!---->
     
    <!---->
     
    <div
      class="well-segment d-flex align-items-center"
    >
      <gl-icon-stub
        class="mr-2 d-none d-sm-block"
        name="clock"
        size="16"
      />
       
      <gl-sprintf-stub
        message="Published to the repository at %{timestamp}"
      />
    </div>
  </div>
</div>
`;

exports[`PackageActivity render to match the default snapshot when there is a pipeline 1`] = `
<div
  class="mb-3"
>
  <h3
    class="gl-font-size-16"
  >
    Activity
  </h3>
   
  <div
    class="info-well"
  >
    <div
      class="well-segment"
    >
      <div
        class="d-flex align-items-center"
      >
        <gl-icon-stub
          class="d-none d-sm-block"
          name="commit"
          size="16"
        />
         
        <!---->
         
        <gl-link-stub
          href="../../commit/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
        >
          xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
        </gl-link-stub>
         
        <clipboard-button-stub
          cssclass="border-0 text-secondary py-0"
          text="xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
          title="Copy commit SHA"
          tooltipplacement="top"
        />
      </div>
       
      <!---->
    </div>
     
    <div
      class="well-segment"
    >
      <div
        class="d-flex align-items-center"
      >
        <gl-icon-stub
          class="mr-2 d-none d-sm-block"
          name="pipeline"
          size="16"
        />
         
        <gl-sprintf-stub
          message="Pipeline %{linkStart}%{linkEnd} triggered %{timestamp} by %{author}"
        />
      </div>
    </div>
     
    <div
      class="well-segment d-flex align-items-center"
    >
      <gl-icon-stub
        class="mr-2 d-none d-sm-block"
        name="clock"
        size="16"
      />
       
      <gl-sprintf-stub
        message="Published to the repository at %{timestamp}"
      />
    </div>
  </div>
</div>
`;
